### PR TITLE
Fix generatedLanguage switch in all plugins #572

### PR DESCRIPTION
**Pull request by jxnu-liguobin**
#572

### CODE REVIEW

**Comment by kobylynskyi** [comment]
@jxnu-liguobin please change target branch to **develop**

**Review by kobylynskyi — CHANGES_REQUESTED** [review]

File: plugins/gradle/graphql-java-codegen-gradle-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/gradle/GraphQLCodegenGradleTask.java [hunk 1]
```diff
@@ -163,15 +145,18 @@ public void generate() throws Exception {
     }
 
     private GraphQLCodegen instantiateCodegen(MappingConfig mappingConfig) throws IOException {
-        switch (generatedLanguage) {
+        java.util.Optional<MappingConfigSupplier> mappingConfigSupplier = buildJsonSupplier();
```
[the remark this answers is not in the document]
Comment (jxnu-liguobin) [reply plugins/gradle/graphql-java-codegen-gradle-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/gradle/GraphQLCodegenGradleTask.java:148]: There is an optional in the this scope. Java doesn't seem to support import rewriting.

File: plugins/maven/graphql-java-codegen-maven-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/GraphQLCodegenMojo.java [hunk 2]
```diff
@@ -261,15 +244,18 @@ public void execute() throws MojoExecutionException {
     }
 
     private GraphQLCodegen instantiateCodegen(MappingConfig mappingConfig) throws IOException {
-        switch (generatedLanguage) {
+        java.util.Optional<MappingConfigSupplier> mappingConfigSupplier = buildJsonSupplier(jsonConfigurationFile);
+        GeneratedLanguage language = mappingConfigSupplier.map(c -> c.get().getGeneratedLanguage() == null ?
+                GeneratedLanguage.JAVA : c.get().getGeneratedLanguage()).orElse(generatedLanguage);
```
Comment (kobylynskyi) [inline plugins/maven/graphql-java-codegen-maven-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/GraphQLCodegenMojo.java:249]:
```suggestion
        GeneratedLanguage language = mappingConfigSupplier.map(Supplier::get)
            .map(MappingConfig::getGeneratedLanguage).orElse(generatedLanguage);
```

File: plugins/gradle/graphql-java-codegen-gradle-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/gradle/GraphQLCodegenGradleTask.java [hunk 3]
```diff
@@ -163,15 +145,18 @@ public void generate() throws Exception {
...
+        GeneratedLanguage language = mappingConfigSupplier.map(c -> c.get().getGeneratedLanguage() == null ? GeneratedLanguage.JAVA
+                : c.get().getGeneratedLanguage()).orElse(generatedLanguage);
```
Comment (kobylynskyi) [inline plugins/gradle/graphql-java-codegen-gradle-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/gradle/GraphQLCodegenGradleTask.java:150]:
```suggestion
        GeneratedLanguage language = mappingConfigSupplier.map(Supplier::get)
            .map(MappingConfig::getGeneratedLanguage).orElse(generatedLanguage);
```

File: plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenPlugin.scala [hunk 4]
```diff
@@ -202,18 +202,20 @@ class GraphQLCodegenPlugin(configuration: Configuration, private[codegen] val co
       }, graphqlCodegen := {
         sLog.value.info(s"Generating files: ${BuildInfo.toString}")
         val mappingConfigSupplier = buildJsonSupplier(jsonConfigurationFile.value.orNull)
+        val language = mappingConfigSupplier.map(m => if (m.get.getGeneratedLanguage == null) GeneratedLanguage.JAVA
```
Comment (kobylynskyi) [inline plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenPlugin.scala:205]: Can you please do the same code I suggested in Gradle/Maven plugins? For me, it seems a bit cleaner. Let me know what you think.

**Review by kobylynskyi — CHANGES_REQUESTED** [review]

File: plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenPlugin.scala (hunk 4 above)
Comment (jxnu-liguobin) [reply plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenPlugin.scala:205]: The function interface looks better .

**Review by kobylynskyi — APPROVED** [review]

**Pull request merged**